src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -0,0 +1,170 @@
+        :param parameters: not really sure what those can be(!)
+        :return: only what is returned by the decorator
+        """
+        logging.info("Running Workflow updater injector poller algorithm...")
+        try:
+            # retrieve list of workflows with unfinished Production
+            # or Processing subscriptions
+            wflowSpecs = self.listActiveWflows.execute()
+            if not wflowSpecs:
+                logging.info("Agent has no active workflows at the moment")
+                return
+
+            # figure out workflows that have pileup
+            puWflows = self.findWflowsWithPileup(wflowSpecs)
+
+            # otherwise, move on retrieving pileups
+            pileupMap = self.getPileupDocs()
+        except Exception as ex:
+            msg = "Caught unexpected exception in WorkflowUpdater. Details:\n%s" % str(ex)
+            logging.exception(msg)
+            raise WorkflowUpdaterException(msg) from None
+
+    def getPileupDocs(self):
+        """
+        Fetch all pileup documents from MSPileup and preprocess the data.
+        :return: a list of dictionaries in the following format:
+          {"pileupName": string with pileup name,
+           "customName": string with custom pileup name - if any,
+           "rses": list of RSE names,
+           "blocks": list of block names}
+        """
+        mgr = RequestHandler()
+        headers = {'Content-Type': 'application/json'}
+        data = mgr.getdata(self.msPileupUrl, params={}, headers=headers, verb='POST',
+                           ckey=self.userKey, cert=self.userCert, encode=True, decode=True)
+        if data and data.get("result", []):
+            if "error" in data["result"][0]:
+                msg = f"Failed to retrieve MSPileup documents. Error: {data}"
+                raise WorkflowUpdaterException(msg)
+
+        logging.info("A total of %d pileup documents have been retrieved.", len(data["result"]))
+        pileupMap = []
+        for puItem in data["result"]:
+            logging.info("Pileup: %s, custom name: %s, expected at: %s, but currently available at: %s",
+                         puItem['pileupName'], puItem['customName'],
+                         puItem['expectedRSEs'], puItem['currentRSEs'])
+            thisPU = {"pileupName": puItem['pileupName'],
+                      "customName": puItem['customName'],
+                      "rses": puItem['currentRSEs'],
+                      "blocks": []}
+            pileupMap.append(thisPU)
+        return pileupMap
+
+    def findWflowsWithPileup(self, listSpecs):
+        """
+        Given a list of workflow names and their respective specs, load each
+        one of them and filter out those that don't require any pileup dataset.
+        :param listSpecs: a list of dictionary with workflow name and spec path
+        :return: a list with the workflow names that require pileup
+        """
+        wflowsWithPU = []
```

Comment:
If we consume it once, then I agree that a generator would make more sense. 
I didn't want to adopt it because then even a "print(f"Found {len(wflowsWithPU)} workflows blah" would defeat the purpose of a generator object.

As an extra background information, this is a list of unique workflow names (strings) running in a given agent, something like < 1k names of up to 155 characters.

Reply:
Based on your numbers the list will be 8.8KB and would be copied from place to place. And, you can achieve len(gen) simply adding another local counter variable. As I said I'm not against the list but I have no idea how it will be used afterwards and I'm trying to estimate the code of having large list go through multiple function calls.